go/adbc/driver/bigquery/statement.go

```diff
@@ -0,0 +1,490 @@
+			Msg:  fmt.Sprintf("unknown statement int type option `%s`", key),
+		}
+	}
+}
+
+func (st *statement) GetOptionDouble(key string) (float64, error) {
+	switch key {
+	default:
+		val, err := st.ConnectionImpl.GetOptionDouble(key)
+		if err == nil {
+			return val, nil
+		}
+		return 0, adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("unknown statement double type option `%s`", key),
+		}
+	}
+}
+
+func (st *statement) setQueryConfig(key, v string) (bool, error) {
+	switch key {
+	case OptionStringQueryDestinationTable:
+		val, err := stringToTable(v)
+		if err == nil {
+			st.queryConfig.Dst = val
+		} else {
+			return true, err
+		}
+	case OptionStringQueryDefaultProjectID:
+		st.queryConfig.DefaultProjectID = v
+	case OptionStringQueryDefaultDatasetID:
+		st.queryConfig.DefaultDatasetID = v
+	case OptionStringQueryCreateDisposition:
+		val, err := stringToTableCreateDisposition(v)
+		if err == nil {
+			st.queryConfig.CreateDisposition = val
+		} else {
+			return true, err
+		}
+	case OptionStringQueryWriteDisposition:
+		val, err := stringToTableWriteDisposition(v)
+		if err == nil {
+			st.queryConfig.WriteDisposition = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryDisableQueryCache:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.DisableQueryCache = val
+		} else {
+			return true, err
+		}
+	case OptionBoolDisableFlattenedResults:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.DisableFlattenedResults = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryAllowLargeResults:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.AllowLargeResults = val
+		} else {
+			return true, err
+		}
+	case OptionStringQueryPriority:
+		val, err := stringToQueryPriority(v)
+		if err == nil {
+			st.queryConfig.Priority = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryUseLegacySQL:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.UseLegacySQL = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryDryRun:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.DryRun = val
+		} else {
+			return true, err
+		}
+	case OptionBoolQueryCreateSession:
+		val, err := stringToBool(v)
+		if err == nil {
+			st.queryConfig.CreateSession = val
+		} else {
+			return true, err
+		}
+
+	default:
+		return false, nil
+	}
+	return true, nil
+}
+
+func (st *statement) setQueryConfigInt(key string, value int64) (bool, error) {
+	switch key {
+	case OptionIntQueryMaxBillingTier:
+		st.queryConfig.MaxBillingTier = int(value)
+	case OptionIntQueryMaxBytesBilled:
+		st.queryConfig.MaxBytesBilled = value
+	case OptionIntQueryJobTimeout:
+		st.queryConfig.JobTimeout = time.Duration(value) * time.Millisecond
+	default:
+		return false, nil
+	}
+	return true, nil
+}
+
+func (st *statement) SetOptions(options map[string]string) error {
+	for k, v := range options {
+		v := v // copy into loop scope
```

Comment:
i don't think you need to do this because you're not doing anything concurrently here, there's no race conditions nor is anything taking the address of the value. right?